packages/markers/src/browser/problem/problem-widget.tsx

```diff
@@ -22,12 +22,17 @@ import { MarkerInfoNode, MarkerNode, MarkerRootNode } from '../marker-tree';
 import { TreeWidget, TreeProps, ContextMenuRenderer, TreeNode, NodeProps, TreeModel } from '@theia/core/lib/browser';
 import { DiagnosticSeverity } from 'vscode-languageserver-types';
 import * as React from 'react';
+import { SelectionService } from '@theia/core/lib/common';
+import { TreeWidgetSelection } from '@theia/core/lib/browser/tree/tree-widget-selection';
 
 export const PROBLEMS_WIDGET_ID = 'problems';
 
 @injectable()
 export class ProblemWidget extends TreeWidget {
 
+    @inject(SelectionService)
+    protected readonly selectionService: SelectionService;
```

Comment:
@akosyakov do you mind taking a look?
I was unsure why we had to inject the `SelectionService` and update the global selection when I was lead to believe it was already handled by the `TreeWidget`. Without this code, the `problem-widget` selection was always `undefined`.

Reply:
updating global selection is a tricky business, you should make sure that you do it only when a widget has focus and remove otherwise

luckily `TreeWidget` does it already properly just set `globalSelection` as `true`: https://github.com/fangnx/theia/blob/b6e36267c781559e52d1d20e30b8a1534a5ddc62/packages/core/src/browser/tree/tree-widget.tsx#L91

Reply:
please revert changes to this file, `SelectionService` is already declared in the super class

Reply:
Yep sure. Sorry that I forgot checking this file ;)